File: lib/core/theme/app_theme_data.dart
// theme_model.dart
import 'package:flutter/material.dart';

/// 主题配置模型
class AppThemeData {
  final String name;
  final Color primaryColor;
  final Color secondaryColor;
  final Color accentColor;
  final bool isCustom;

  const AppThemeData({
    required this.name,
    required this.primaryColor,
    required this.secondaryColor,
    required this.accentColor,
    this.isCustom = false,
  });

  // ============ 私有颜色常量 ============
  // 背景色
  static const Color _backgroundColor = Color(0xFFF7F8FA);
  static const Color _surfaceColor = Colors.white;
  static const Color _darkBackgroundColor = Color(0xFF0F172A);
  static const Color _darkSurfaceColor = Color(0xFF111827);

  // 文字颜色
  static const Color _textPrimaryColor = Color(0xFF1F2937);
  static const Color _textSecondaryColor = Color(0xFF4B5563);
  static const Color _textDarkPrimaryColor = Color(0xFFE5E7EB);
  static const Color _textDarkSecondaryColor = Color(0xFF9CA3AF);

  // 边框和分割线
  static const Color _borderColor = Color(0xFFE2E8F0);
  static const Color _darkBorderColor = Color(0xFF1F2937);

  // ============ 布局常量 ============
  // 圆角
  static const double borderRadiusSmall = 8.0;
  static const double borderRadiusMedium = 12.0;
  static const double borderRadiusLarge = 16.0;

  // 间距
  static const double spacingSmall = 8.0;
  static const double spacingMedium = 16.0;
  static const double spacingLarge = 24.0;
  static const double spacingXLarge = 32.0;

  // 侧边栏宽度（紧凑型）
  static const double sidebarExpandedWidth = 200.0;
  static const double sidebarCollapsedWidth = 60.0;

  // 动画时长
  static const Duration animationDuration = Duration(milliseconds: 300);

  /// 预设主题列表（你原有的主题作为第一个）
  static final List<AppThemeData> presetThemes = [
    // 默认主题（保留你原有的设计）
    const AppThemeData(
      name: '默认紫',
      primaryColor: Color(0xFF6C5CE7),
      secondaryColor: Color(0xFFA29BFE),
      accentColor: Color(0xFF00B894),
    ),
    const AppThemeData(
      name: '圣诞红',
      primaryColor: Color(0xFFD32F2F),
      secondaryColor: Color(0xFFEF5350),
      accentColor: Color(0xFFFF9800),
    ),
    const AppThemeData(
      name: '海洋蓝',
      primaryColor: Color(0xFF0277BD),
      secondaryColor: Color(0xFF4FC3F7),
      accentColor: Color(0xFF00BCD4),
    ),
    const AppThemeData(
      name: '自然绿',
      primaryColor: Color(0xFF388E3C),
      secondaryColor: Color(0xFF66BB6A),
      accentColor: Color(0xFF8BC34A),
    ),
    const AppThemeData(
      name: '温暖橙',
      primaryColor: Color(0xFFE64A19),
      secondaryColor: Color(0xFFFF7043),
      accentColor: Color(0xFFFFB74D),
    ),
    const AppThemeData(
      name: '优雅紫',
      primaryColor: Color(0xFF7B1FA2),
      secondaryColor: Color(0xFFAB47BC),
      accentColor: Color(0xFFBA68C8),
    ),
  ];

  /// 生成浅色主题（保留你的设计风格）
  ThemeData generateLightTheme() {
    return ThemeData(
      useMaterial3: true,
      brightness: Brightness.light,
      primaryColor: primaryColor,
      scaffoldBackgroundColor: _backgroundColor,
      colorScheme: ColorScheme.light(
        primary: primaryColor,
        secondary: secondaryColor,
        surface: _surfaceColor,
        error: Colors.red,
        onPrimary: getContrastColor(primaryColor),
        onSecondary: getContrastColor(secondaryColor),
        onSurface: _textPrimaryColor,
        onError: Colors.white,
        surfaceContainerHighest: const Color(0xFFF1F5F9),
        outline: const Color(0xFF94A3B8),
        outlineVariant: _borderColor,
        onSurfaceVariant: _textSecondaryColor,
      ),
      appBarTheme: const AppBarTheme(
        elevation: 0,
        backgroundColor: _surfaceColor,
        foregroundColor: _textPrimaryColor,
        centerTitle: false,
        titleTextStyle: TextStyle(
          color: _textPrimaryColor,
          fontSize: 20,
          fontWeight: FontWeight.w600,
        ),
      ),
      cardTheme: CardThemeData(
        elevation: 0,
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(borderRadiusMedium),
          side: const BorderSide(color: _borderColor, width: 1),
        ),
        color: _surfaceColor,
      ),
      elevatedButtonTheme: ElevatedButtonThemeData(
        style: ElevatedButton.styleFrom(
          elevation: 0,
          backgroundColor: primaryColor,
          foregroundColor: getContrastColor(primaryColor),
          padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 12),
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(borderRadiusSmall),
          ),
        ),
      ),
      outlinedButtonTheme: OutlinedButtonThemeData(
        style: OutlinedButton.styleFrom(
          elevation: 0,
          backgroundColor: _surfaceColor,
          foregroundColor: primaryColor,
          padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 12),
          side: const BorderSide(color: _borderColor, width: 1),
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(borderRadiusLarge),
          ),
        ),
      ),
      inputDecorationTheme: InputDecorationTheme(
        filled: true,
        fillColor: const Color(0xFFF8FAFC),
        border: OutlineInputBorder(
          borderRadius: BorderRadius.circular(borderRadiusSmall),
          borderSide: const BorderSide(color: _borderColor),
        ),
        enabledBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(borderRadiusSmall),
          borderSide: const BorderSide(color: _borderColor),
        ),
        focusedBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(borderRadiusSmall),
          borderSide: BorderSide(color: primaryColor, width: 1.2),
        ),
        errorBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(borderRadiusSmall),
          borderSide: const BorderSide(color: Colors.redAccent),
        ),
        focusedErrorBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(borderRadiusSmall),
          borderSide: const BorderSide(color: Colors.redAccent, width: 1.2),
        ),
        labelStyle: const TextStyle(color: _textSecondaryColor),
        hintStyle: const TextStyle(color: _textSecondaryColor),
      ),
      dividerTheme: const DividerThemeData(
        color: _borderColor,
        thickness: 1,
      ),
      iconTheme: const IconThemeData(color: _textSecondaryColor),
      floatingActionButtonTheme: FloatingActionButtonThemeData(
        backgroundColor: primaryColor,
        foregroundColor: getContrastColor(primaryColor),
      ),
      textTheme: const TextTheme(
        displayLarge: TextStyle(
          fontSize: 32,
          fontWeight: FontWeight.bold,
          color: _textPrimaryColor,
        ),
        displayMedium: TextStyle(
          fontSize: 28,
          fontWeight: FontWeight.bold,
          color: _textPrimaryColor,
        ),
        displaySmall: TextStyle(
          fontSize: 24,
          fontWeight: FontWeight.w600,
          color: _textPrimaryColor,
        ),
        headlineMedium: TextStyle(
          fontSize: 20,
          fontWeight: FontWeight.w600,
          color: _textPrimaryColor,
        ),
        bodyLarge: TextStyle(fontSize: 16, color: _textPrimaryColor),
        bodyMedium: TextStyle(fontSize: 14, color: _textSecondaryColor),
        bodySmall: TextStyle(fontSize: 12, color: _textSecondaryColor),
        labelLarge: TextStyle(fontSize: 14, color: _textSecondaryColor),
        labelMedium: TextStyle(fontSize: 12, color: _textSecondaryColor),
        labelSmall: TextStyle(fontSize: 11, color: _textSecondaryColor),
      ),
    );
  }

  /// 生成深色主题（保留你的设计风格）
  ThemeData generateDarkTheme() {
    return ThemeData(
      useMaterial3: true,
      brightness: Brightness.dark,
      primaryColor: primaryColor,
      scaffoldBackgroundColor: _darkBackgroundColor,
      colorScheme: ColorScheme.dark(
        primary: primaryColor,
        secondary: secondaryColor,
        surface: _darkSurfaceColor,
        error: Colors.redAccent,
        onPrimary: getContrastColor(primaryColor), // 智能对比色
        onSecondary: getContrastColor(secondaryColor),
        onSurface: _textDarkPrimaryColor,
        onError: Colors.white,
        surfaceContainerHighest: const Color(0xFF1F2937),
        outline: const Color(0xFF475569),
        outlineVariant: _darkBorderColor,
        onSurfaceVariant: _textDarkSecondaryColor,
      ),
      appBarTheme: const AppBarTheme(
        elevation: 0,
        backgroundColor: _darkSurfaceColor,
        foregroundColor: _textDarkPrimaryColor,
        centerTitle: false,
        titleTextStyle: TextStyle(
          color: _textDarkPrimaryColor,
          fontSize: 20,
          fontWeight: FontWeight.w600,
        ),
      ),
      cardTheme: CardThemeData(
        elevation: 0,
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(borderRadiusMedium),
          side: const BorderSide(color: _darkBorderColor, width: 1),
        ),
        color: _darkSurfaceColor,
      ),
      elevatedButtonTheme: ElevatedButtonThemeData(
        style: ElevatedButton.styleFrom(
          elevation: 0,
          backgroundColor: primaryColor,
          foregroundColor: getContrastColor(primaryColor),
          padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 12),
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(borderRadiusSmall),
          ),
        ),
      ),
      outlinedButtonTheme: OutlinedButtonThemeData(
        style: OutlinedButton.styleFrom(
          elevation: 0,
          backgroundColor: _darkSurfaceColor,
          foregroundColor: primaryColor,
          padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 12),
          side: const BorderSide(color: _darkBorderColor, width: 1),
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(borderRadiusLarge),
          ),
        ),
      ),
      inputDecorationTheme: InputDecorationTheme(
        filled: true,
        fillColor: const Color(0xFF111827),
        border: OutlineInputBorder(
          borderRadius: BorderRadius.circular(borderRadiusSmall),
          borderSide: const BorderSide(color: _darkBorderColor),
        ),
        enabledBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(borderRadiusSmall),
          borderSide: const BorderSide(color: _darkBorderColor),
        ),
        focusedBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(borderRadiusSmall),
          borderSide: BorderSide(color: primaryColor, width: 1.2),
        ),
        errorBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(borderRadiusSmall),
          borderSide: const BorderSide(color: Colors.redAccent),
        ),
        focusedErrorBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(borderRadiusSmall),
          borderSide: const BorderSide(color: Colors.redAccent, width: 1.2),
        ),
        labelStyle: const TextStyle(color: _textDarkSecondaryColor),
        hintStyle: const TextStyle(color: _textDarkSecondaryColor),
      ),
      dividerTheme: const DividerThemeData(
        color: _darkBorderColor,
        thickness: 1,
      ),
      iconTheme: const IconThemeData(color: _textDarkSecondaryColor),
      floatingActionButtonTheme: FloatingActionButtonThemeData(
        backgroundColor: primaryColor,
        foregroundColor: getContrastColor(primaryColor),
      ),
      textTheme: const TextTheme(
        displayLarge: TextStyle(
          fontSize: 32,
          fontWeight: FontWeight.bold,
          color: _textDarkPrimaryColor,
        ),
        displayMedium: TextStyle(
          fontSize: 28,
          fontWeight: FontWeight.bold,
          color: _textDarkPrimaryColor,
        ),
        displaySmall: TextStyle(
          fontSize: 24,
          fontWeight: FontWeight.w600,
          color: _textDarkPrimaryColor,
        ),
        headlineMedium: TextStyle(
          fontSize: 20,
          fontWeight: FontWeight.w600,
          color: _textDarkPrimaryColor,
        ),
        bodyLarge: TextStyle(fontSize: 16, color: _textDarkPrimaryColor),
        bodyMedium: TextStyle(fontSize: 14, color: _textDarkSecondaryColor),
        bodySmall: TextStyle(fontSize: 12, color: _textDarkSecondaryColor),
        labelLarge: TextStyle(fontSize: 14, color: _textDarkSecondaryColor),
        labelMedium: TextStyle(fontSize: 12, color: _textDarkSecondaryColor),
        labelSmall: TextStyle(fontSize: 11, color: _textDarkSecondaryColor),
      ),
    );
  }

  /// 计算对比色（确保按钮文字可见）
  static Color getContrastColor(Color color) {
    final luminance = color.computeLuminance();
    return luminance > 0.5 ? Colors.black87 : Colors.white;
  }

  static Color getBorderColor(ThemeData theme) {
    return theme.brightness == Brightness.light
        ? _borderColor
        : _darkBorderColor;
  }

  static Color getTextColor(ThemeData theme, {bool isPrimary = true}) {
    return theme.brightness == Brightness.light
        ? (isPrimary ? _textPrimaryColor : _textSecondaryColor)
        : (isPrimary ? _textDarkPrimaryColor : _textDarkSecondaryColor);
  }

  /// 转换为Map（用于持久化存储）
  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'primaryColor': primaryColor.toARGB32(),
      'secondaryColor': secondaryColor.toARGB32(),
      'accentColor': accentColor.toARGB32(),
      'isCustom': isCustom,
    };
  }

  /// 从Map创建（用于持久化存储）
  factory AppThemeData.fromJson(Map<String, dynamic> json) {
    return AppThemeData(
      name: json['name'] as String,
      primaryColor: Color(json['primaryColor'] as int),
      secondaryColor: Color(json['secondaryColor'] as int),
      accentColor: Color(json['accentColor'] as int),
      isCustom: json['isCustom'] as bool? ?? false,
    );
  }

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;
    return other is AppThemeData &&
        other.name == name &&
        other.primaryColor == primaryColor &&
        other.secondaryColor == secondaryColor &&
        other.accentColor == accentColor &&
        other.isCustom == isCustom;
  }

  @override
  int get hashCode =>
      Object.hash(name, primaryColor, secondaryColor, accentColor, isCustom);
}

File: src/core/theme/app-theme-data.ts
import { LocalizationKeys } from '../localization/localization-keys';
import { getContrastColor, lerpColor, withAlpha } from './color-utils';

/**
 * 主题配置模型
 *
 * 对应 Flutter 版 `lib/core/theme/app_theme_data.dart`。
 * 桌面版通过 `generateLightTheme/generateDarkTheme` 产出 `ThemeData`；
 * Web 版改为产出一组 CSS 自定义属性（design tokens），由 `applyThemeTokens` 注入 :root。
 */
export interface AppThemeData {
  /** 主题名称（自定义主题用） */
  name: string;
  /** 预设主题的国际化 key，存在时优先用于展示 */
  localizationKey?: string;
  primaryColor: string;
  secondaryColor: string;
  accentColor: string;
  isCustom: boolean;
}

// ============ 私有颜色常量（与 Flutter 版逐值对齐）============
const BACKGROUND_COLOR = '#F5F6FA';
const SURFACE_COLOR = '#FFFFFF';
const DARK_BACKGROUND_COLOR = '#1A1A2E';
const DARK_SURFACE_COLOR = '#16213E';

const TEXT_PRIMARY_COLOR = '#2D3436';
const TEXT_SECONDARY_COLOR = '#636E72';
const TEXT_DARK_PRIMARY_COLOR = '#DFE6E9';
const TEXT_DARK_SECONDARY_COLOR = '#B2BEC3';

const BORDER_COLOR = '#DFE6E9';
const DARK_BORDER_COLOR = '#2D3436';

// ============ 布局常量 ============
export const ThemeMetrics = {
  borderRadiusSmall: 8,
  borderRadiusMedium: 12,
  borderRadiusLarge: 16,

  spacingSmall: 8,
  spacingMedium: 16,
  spacingLarge: 24,
  spacingXLarge: 32,

  sidebarExpandedWidth: 200,
  sidebarCollapsedWidth: 60,

  /** 动画时长（毫秒） */
  animationDuration: 300,

  /** 桌面布局最小内容宽度，低于此值横向滚动 */
  desktopMinContentWidth: 1000,
} as const;

/** 主题模式，等价于 Flutter 的 ThemeMode */
export type ThemeMode = 'system' | 'light' | 'dark';
export const THEME_MODES: ThemeMode[] = ['system', 'light', 'dark'];

/** 预设主题列表 */
export const presetThemes: AppThemeData[] = [
  {
    name: '默认紫',
    localizationKey: LocalizationKeys.themeDefaultPurple,
    primaryColor: '#6C5CE7',
    secondaryColor: '#A29BFE',
    accentColor: '#00B894',
    isCustom: false,
  },
  {
    name: '圣诞红',
    localizationKey: LocalizationKeys.themeChristmasRed,
    primaryColor: '#D32F2F',
    secondaryColor: '#EF5350',
    accentColor: '#FF9800',
    isCustom: false,
  },
  {
    name: '海洋蓝',
    localizationKey: LocalizationKeys.themeOceanBlue,
    primaryColor: '#0277BD',
    secondaryColor: '#4FC3F7',
    accentColor: '#00BCD4',
    isCustom: false,
  },
  {
    name: '自然绿',
    localizationKey: LocalizationKeys.themeNaturalGreen,
    primaryColor: '#388E3C',
    secondaryColor: '#66BB6A',
    accentColor: '#8BC34A',
    isCustom: false,
  },
  {
    name: '温暖橙',
    localizationKey: LocalizationKeys.themeWarmOrange,
    primaryColor: '#E64A19',
    secondaryColor: '#FF7043',
    accentColor: '#FFB74D',
    isCustom: false,
  },
  {
    name: '优雅紫',
    localizationKey: LocalizationKeys.themeElegantPurple,
    primaryColor: '#7B1FA2',
    secondaryColor: '#AB47BC',
    accentColor: '#BA68C8',
    isCustom: false,
  },
];

/** 主题相等判断（对应 Flutter 的 operator ==） */
export function isSameTheme(a: AppThemeData, b: AppThemeData): boolean {
  return (
    a.name === b.name &&
    a.primaryColor === b.primaryColor &&
    a.secondaryColor === b.secondaryColor &&
    a.accentColor === b.accentColor &&
    a.isCustom === b.isCustom
  );
}

/** 生成的一套完整令牌 */
export type ThemeTokens = Record<string, string>;

/**
 * 生成主题令牌
 *
 * @param theme      配色方案
 * @param brightness 明/暗
 * @param adjustment 对比度调整 0.0–1.0
 *                   - light: 背景/表面向黑色插值（数值越大越暗）
 *                   - dark : 背景向 #050505、表面向 #0A0A15 插值
 */
export function generateThemeTokens(
  theme: AppThemeData,
  brightness: 'light' | 'dark',
  adjustment = 0,
): ThemeTokens {
  const isDark = brightness === 'dark';

  const bgColor = isDark
    ? lerpColor(DARK_BACKGROUND_COLOR, '#050505', adjustment)
    : lerpColor(BACKGROUND_COLOR, '#000000', adjustment);
  const surfaceColor = isDark
    ? lerpColor(DARK_SURFACE_COLOR, '#0A0A15', adjustment)
    : lerpColor(SURFACE_COLOR, '#000000', adjustment);

  const textPrimary = isDark ? TEXT_DARK_PRIMARY_COLOR : TEXT_PRIMARY_COLOR;
  const textSecondary = isDark ? TEXT_DARK_SECONDARY_COLOR : TEXT_SECONDARY_COLOR;
  const border = isDark ? DARK_BORDER_COLOR : BORDER_COLOR;

  const { primaryColor, secondaryColor, accentColor } = theme;

  return {
    // ── 配色方案 ──
    '--color-primary': primaryColor,
    '--color-secondary': secondaryColor,
    '--color-accent': accentColor,
    '--color-on-primary': getContrastColor(primaryColor),
    '--color-on-secondary': getContrastColor(secondaryColor),

    '--color-background': bgColor,
    '--color-surface': surfaceColor,
    '--color-on-surface': isDark ? '#FFFFFF' : 'rgba(0, 0, 0, 0.87)',
    '--color-surface-container-highest': isDark ? '#2C2C2C' : '#F5F5F5',
    '--color-outline': withAlpha(primaryColor, isDark ? 0.4 : 0.3),

    '--color-error': isDark ? '#FF5252' : '#F44336',
    '--color-on-error': '#FFFFFF',
    '--color-success': '#00B894',
    '--color-warning': '#FF9800',
    '--color-danger': '#E53935',
    '--color-caution': '#FDD835',

    // ── 文字 ──
    '--color-text-primary': textPrimary,
    '--color-text-secondary': textSecondary,

    // ── 边框 ──
    '--color-border': border,

    // ── 常用透明度衍生（避免组件里反复算）──
    '--color-primary-a10': withAlpha(primaryColor, 0.1),
    '--color-primary-a20': withAlpha(primaryColor, 0.2),
    '--color-primary-a30': withAlpha(primaryColor, 0.3),
    '--color-on-surface-a04': withAlpha(isDark ? '#FFFFFF' : '#000000', 0.04),
    '--color-on-surface-a08': withAlpha(isDark ? '#FFFFFF' : '#000000', 0.08),
    '--color-on-surface-a40': withAlpha(isDark ? '#FFFFFF' : '#000000', 0.4),
    '--color-on-surface-a60': withAlpha(isDark ? '#FFFFFF' : '#000000', 0.6),

    // ── 阴影 ──
    '--shadow-sm': isDark
      ? '0 1px 4px rgba(0, 0, 0, 0.45)'
      : '0 1px 4px rgba(0, 0, 0, 0.1)',
    '--shadow-md': isDark
      ? '0 4px 16px rgba(0, 0, 0, 0.5)'
      : '0 4px 16px rgba(0, 0, 0, 0.12)',
    '--shadow-lg': isDark
      ? '0 8px 32px rgba(0, 0, 0, 0.6)'
      : '0 8px 32px rgba(0, 0, 0, 0.16)',

    // ── 布局令牌 ──
    '--radius-sm': `${ThemeMetrics.borderRadiusSmall}px`,
    '--radius-md': `${ThemeMetrics.borderRadiusMedium}px`,
    '--radius-lg': `${ThemeMetrics.borderRadiusLarge}px`,
    '--space-sm': `${ThemeMetrics.spacingSmall}px`,
    '--space-md': `${ThemeMetrics.spacingMedium}px`,
    '--space-lg': `${ThemeMetrics.spacingLarge}px`,
    '--space-xl': `${ThemeMetrics.spacingXLarge}px`,
    '--sidebar-expanded': `${ThemeMetrics.sidebarExpandedWidth}px`,
    '--sidebar-collapsed': `${ThemeMetrics.sidebarCollapsedWidth}px`,
    '--anim-duration': `${ThemeMetrics.animationDuration}ms`,
  };
}

/** 把令牌写入 <html> 的 style，并打上 data-brightness 标记 */
export function applyThemeTokens(tokens: ThemeTokens, brightness: 'light' | 'dark'): void {
  const root = document.documentElement;
  for (const [key, value] of Object.entries(tokens)) {
    root.style.setProperty(key, value);
  }
  root.dataset.brightness = brightness;
  root.style.colorScheme = brightness;
}

/** 序列化（持久化用），对应 Flutter 的 toJson */
export function themeToJson(theme: AppThemeData): Record<string, unknown> {
  return {
    name: theme.name,
    localizationKey: theme.localizationKey,
    primaryColor: theme.primaryColor,
    secondaryColor: theme.secondaryColor,
    accentColor: theme.accentColor,
    isCustom: theme.isCustom,
  };
}

/** 反序列化，容错回退到默认主题 */
export function themeFromJson(json: unknown): AppThemeData {
  const fallback = presetThemes[0];
  if (!json || typeof json !== 'object') return fallback;
  const raw = json as Record<string, unknown>;
  const pick = (key: string, def: string) =>
    typeof raw[key] === 'string' ? (raw[key] as string) : def;
  return {
    name: pick('name', fallback.name),
    localizationKey:
      typeof raw.localizationKey === 'string' ? raw.localizationKey : undefined,
    primaryColor: pick('primaryColor', fallback.primaryColor),
    secondaryColor: pick('secondaryColor', fallback.secondaryColor),
    accentColor: pick('accentColor', fallback.accentColor),
    isCustom: raw.isCustom === true,
  };
}

/** 主题模式对应的 Material Symbols 图标名 */
export function themeModeIcon(mode: ThemeMode): string {
  switch (mode) {
    case 'system':
      return 'brightness_auto';
    case 'light':
      return 'light_mode';
    case 'dark':
      return 'dark_mode';
  }
}

/** 主题模式对应的国际化 key */
export function themeModeLocalizationKey(mode: ThemeMode): string {
  switch (mode) {
    case 'system':
      return LocalizationKeys.themeModeSystem;
    case 'light':
      return LocalizationKeys.themeModeLight;
    case 'dark':
      return LocalizationKeys.themeModeDark;
  }
}
